astropy/io/fits/util.py

```diff
@@ -882,3 +882,23 @@ def _free_space_check(hdulist, dirname=None):
             hdu._close()
 
         raise OSError(error_message + str(exc))
+
+
+def _extract_number(value, default):
+    """
+    Attempts to extract an integer number from the given value. If the
+    extraction fails, the value of the 'default' argument is returned.
+    """
+
+    try:
```

Comment:
Not really, the `_is_int` basically boils down to type-checking using `isinstance(val, all_integer_types)`.